packages/ui-components/src/components/NativeSelect/NativeSelect.component.tsx

```diff
@@ -0,0 +1,254 @@
+   */
+  invalid?: boolean
+
+  /**
+   * Highlights the select element as valid, indicating correct user input or successful validation.
+   */
+  valid?: boolean
+
+  /**
+   * Displays a loading indicator over the select element.
+   * Used for async operations like fetching data.
+   */
+  loading?: boolean
+
+  /**
+   * Displays an error state over the select element, such as during data fetching errors.
+   * Should not be used for validation errors.
+   */
+  error?: boolean
+
+  /**
+   * Event handler for the change event of the select element.
+   * Triggered when the user changes the selected option.
+   */
+  // eslint-disable-next-line no-unused-vars
+  onChange?: (event: ChangeEvent<HTMLSelectElement>) => void
+
+  /**
+   * Event handler for the click event on the select element.
+   * Triggered when the user clicks on the select element.
+   */
+  // eslint-disable-next-line no-unused-vars
+  onClick?: (event: MouseEvent<HTMLSelectElement>) => void
+
+  /**
+   * Additionl CSS classes to apply to the outer wrapper of the select component for custom styling.
+   */
+  wrapperClassName?: string
+}
+
+/**
+ * Renders the appropriate 'state' icon based on the state of the select element
+ */
+const SelectIcons: FC<{
+  isLoading: boolean
+  hasError: boolean
+  isInvalid: boolean
+  isValid: boolean
+  disabled?: boolean
+}> = ({ isLoading, hasError, isInvalid, isValid, disabled }) => {
+  if (isLoading) {
+    return (
+      <div className={`juno-select-loading ${stateClasses.loading}`}>
+        <Spinner className={stateClasses.loadingSpinner} />
+      </div>
+    )
+  }
+  if (hasError) {
+    return (
+      <div className={`juno-select-error-text ${stateClasses.error}`}>
+        <Icon icon="errorOutline" color="jn-text-theme-error" className={stateClasses.errorIcon} />
+      </div>
+    )
+  }
+  return (
+    <div className={`${iconBaseClasses} ${disabled ? iconDisabledClasses : ""}`}>
+      {isInvalid && <Icon icon="dangerous" color="jn-text-theme-error" />}
+      {isValid && <Icon icon="checkCircle" color="jn-text-theme-success" />}
+      <Icon icon="expandMore" />
+    </div>
+  )
+}
+
+/**
+ * A basic, uncontrolled native HTML select component.
+ * Takes NativeSelectOption and NativeSelectOptionGroup children as options.
+ * Handles states such as loading, valid, invalid, and error.
+ */
+export const NativeSelect: FC<NativeSelectProps> = ({
+  name = "Unnamed Select",
+  id = "",
+  children,
+  className = "",
+  disabled = false,
+  invalid = false,
+  valid = false,
+  loading = false,
+  error = false,
+  onChange,
+  onClick,
+  wrapperClassName = "",
+  ...props
+}) => {
+  const [isLoading, setIsLoading] = useState(loading)
+  const [isValid, setIsValid] = useState(valid)
+  const [isInvalid, setIsInvalid] = useState(invalid)
+  const [hasError, setHasError] = useState(error)
+
+  useEffect(() => setIsLoading(loading), [loading])
```

Comment:
All of this useEffects are so unnecessary I would really put it into consideration of future improvements.

Reply:
Done: https://github.com/cloudoperators/juno/issues/570 (by you). But I added this one as an example.